packages/syft/tests/syft/core/tensor/autodp/row_entity_phi_test.py

```diff
@@ -0,0 +1,125 @@
+# stdlib
+from typing import List
+
+# third party
+import numpy as np
+import pytest
+
+# syft absolute
+from syft import deserialize
+from syft import serialize
+from syft.core.adp.entity import Entity
+from syft.core.tensor.autodp.row_entity_phi import RowEntityPhiTensor as REPT
+from syft.core.tensor.autodp.single_entity_phi import SingleEntityPhiTensor as SEPT
+from syft.core.tensor.tensor import Tensor
+
+################### EQUALITY OPERATORS ###############################################
+
+# Global constants
+ishan = Entity(name="Ishan")
+supreme_leader = Entity(name="Trask")
+dims = np.random.randint(10) + 1  # Avoid size 0
+row_count = np.random.randint(7) + 1
+
+
+@pytest.fixture
+def upper_bound() -> np.ndarray:
+    """This is used to specify the max_vals for a SEPT that is either binary or randomly generated b/w 0-1"""
+    max_values = np.ones(dims)
+    return max_values
+
+
+@pytest.fixture
+def lower_bound() -> np.ndarray:
+    """This is used to specify the min_vals for a SEPT that is either binary or randomly generated b/w 0-1"""
+    min_values = np.zeros(dims)
+    return min_values
+
+
+@pytest.fixture
+def reference_data() -> np.ndarray:
+    """This generates random data to test the equality operators"""
+    reference_data = np.random.random((dims, dims))
+    return reference_data
+
+
+@pytest.fixture
+def row_data(lower_bound: np.ndarray, upper_bound: np.ndarray) -> List:
+    """This generates a random number of SEPTs to populate the REPTs."""
+    reference_data = []
+    for _ in range(row_count):
+        reference_data.append(
+            SEPT(
+                child=np.random.random((dims, dims)),
+                entity=ishan,
+                min_vals=lower_bound,
+                max_vals=upper_bound,
+            )
+        )
+    return reference_data
+
+
+@pytest.fixture
+def reference_binary_data() -> np.ndarray:
+    """Generate binary data to test the equality operators with bools"""
+    binary_data = np.random.randint(2, size=(dims, dims))
+    return binary_data
+
+
+def test_eq(row_data: List) -> bool:
+    """Test equality between two identical RowEntityPhiTensors"""
+    reference_tensor = REPT(rows=row_data)
+    second_tensor = REPT(rows=row_data)
+    third_tensor = reference_tensor
```

Comment:
This assignment will mean that `third_tensor` and `reference_tensor` will point to the same variable in memory and have the same id() so comparing it later should always be True, was that intended?